src/Nethermind/Nethermind.TxPool/Comparison/CompareReplacedBlobTx.cs

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using Nethermind.Core;
+
+namespace Nethermind.TxPool.Comparison;
+
+/// <summary>
+/// Compare fee of newcomer blob transaction with fee of transaction intended to be replaced
+/// </summary>
+public class CompareReplacedBlobTx : IComparer<Transaction?>
+{
+    public static readonly CompareReplacedBlobTx Instance = new();
+
+    private CompareReplacedBlobTx() { }
+
+    // To replace old blob transaction, new transaction needs to have fee at least 2x higher than current fee.
+    // 2x higher must be MaxPriorityFeePerGas, MaxFeePerGas and MaxFeePerDataGas
+    public int Compare(Transaction? newTx, Transaction? oldTx)
+    {
+        if (ReferenceEquals(newTx, oldTx)) return TxComparisonResult.NotDecided;
+        if (ReferenceEquals(null, oldTx)) return TxComparisonResult.KeepOld;
+        if (ReferenceEquals(null, newTx)) return TxComparisonResult.TakeNew;
+
+        // do not allow to replace blob tx by the one with lower number of blobs
+        if (oldTx.BlobVersionedHashes is null || newTx.BlobVersionedHashes is null) return TxComparisonResult.KeepOld;
+        if (oldTx.BlobVersionedHashes.Length > newTx.BlobVersionedHashes.Length) return TxComparisonResult.KeepOld;
```

Comment:
This part is not in-line with Geth, it is based on discussion on one of Cancun Implementers Calls. We can open it for discussion on next Cancun Implementers Call. But on the other hand tx replacing is implementation detail, we have some general assumptions, but don't need to act 100% same way

Reply:
It was discussed on Cancun Implementers Call. This feature is not a part of spec/hard requirement, but is considered as good to have, so it is fine to keep it as it is